src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+  const dateParts = date.split(fromSeparator);
+  const objFrom = {};
+  const arrTo = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    objFrom[fromFormat[i]] = dateParts[i];
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    for (const entry of Object.entries(objFrom)) {
+      const twoLettersTo = toFormat[i].slice(0, 2);
+      const twoLettersFrom = entry[0].slice(0, 2);
+
+      if (toFormat[i] === entry[0]) {
+        arrTo[i] = entry[1];
+      } else if (twoLettersTo === twoLettersFrom) {
+        if (entry[0].length > toFormat[i].length) {
+          arrTo[i] = entry[1].slice(2, 4);
```

Comment:
When slicing the year to convert from 'YYYY' to 'YY', you should use the last two digits. Currently, it slices from the third to the fourth digit, which would not return the correct year in 'YY' format.